### PR TITLE
Improve switching to spectator using team manager

### PR DESCRIPTION
**Pull request by cogito123**
So far switching to spectator was mishandled if player was dead or in alternative controller. I've noticed that players removed by captain were not hidden on map even though all paths should be covered. Killing character right before switching to spectator makes us leave prematurely from routine that initializes spectator state.

### CODE REVIEW

**Comment by cogito123** [comment]
Note that I still have to test it thoroughly. It's a high risk change with lots of corner cases.

File: maps/biter_battles_v2/gui.lua [hunk 1]
```diff
@@ -1008,8 +1008,48 @@ function join_team(player, force_name, forced_join, auto_join)
...
+    -- for a respawn or has alternative controller set.
     if not player.character then
-        return
+        -- Reposition camera to chunk which is guaranteed to exists to avoid
+        -- any issue with character positioning. Repositioning is performed
+        -- only for cases where character is to be created.
+        local origin = { 0, 0 }
+        if player.ticks_to_respawn then
+            -- Character cannot be created while dead, we can only speed up
+            -- respawn and pray it doesn't race.
```
Comment (clifffrey) [inline maps/biter_battles_v2/gui.lua:1020]: I'd suggest changing "pray" to "hope", as some people have strong opinions/reactions to religious rituals

File: maps/biter_battles_v2/gui.lua [hunk 2]
```diff
@@ -1008,8 +1008,48 @@ function join_team(player, force_name, forced_join, auto_join)
...
+            player.teleport(origin)
+            player.ticks_to_respawn = nil
+            if not player.character then
+                log('WARN: spectate: player ' .. player.name .. ' still without a character')
```
Comment (clifffrey) [inline maps/biter_battles_v2/gui.lua:1024]: I wonder if it might be more useful to print out a message to either everyone's game, or to all admins in the game.  That said, I don't feel too strongly about this either way.

**Review by clifffrey — APPROVED** [review]

File: maps/biter_battles_v2/gui.lua (hunk 2 above)
Comment (cogito123) [reply maps/biter_battles_v2/gui.lua:1102]: I tested this a lot and this case never triggered.
